tests/FSharpLint.Core.Tests/Rules/Conventions/EnsureTailCallDiagnosticsInRecursiveFunctions.fs

```diff
@@ -0,0 +1,47 @@
+﻿module FSharpLint.Core.Tests.Rules.Conventions.EnsureTailCallDiagnosticsInRecursiveFunctions
+
+open NUnit.Framework
+
+open FSharpLint.Rules
+
+[<TestFixture>]
+type TestEnsureTailCallDiagnosticsInRecursiveFunctions() =
+    inherit TestAstNodeRuleBase.TestAstNodeRuleBase(EnsureTailCallDiagnosticsInRecursiveFunctions.rule)
+
+    [<Test>]
+    member this.``Should error when function is recursive, but has no [<TailCall>] attribute``() =
+        this.Parse """
+let rec Foo someParam =
+    if someParam then
+        Foo false
+    else
+        ()"""
+
+        Assert.IsTrue <| this.ErrorExistsAt(2, 8)
+
+    [<Test>]
+    member this.``Should not error when function is recursive and has [<TailCall>] attribute``() =
+        this.Parse """
+[<TailCall>]
+let rec Foo someParam =
+    if someParam then
+        Foo false
+    else
+        ()"""
+
+        Assert.IsTrue this.NoErrorsExist
+
+    [<Test>]
+    member this.``Should error when function is defined with rec keyword but is not recursive``() =
+        this.Parse """
+let rec Foo someParam =
+    ()
+
+module Bar =    
+    let Foo = 0"""
+
+        Assert.IsTrue 
+            <| this.ErrorWithMessageExistsAt(
+                """The 'Foo' function has a "rec" keyword, but it is not really recursive, consider removing it.""", 
```

Comment:
@webwarrior-ws I've changed my mind about this test, essentially because if I merge PR#666 then it will give misleading info. So let's change this test to Assert.NoErrorsExist

Reply:
(but don't undo the refactoring commit please, I think it's good.)